### ISSUE TITLE
 Running Integration Tests without Address Sanitizer results in different numbers for Passed/Failed with every test run

### ISSUE THREAD

**Issue by stanislaw** [mull-project/mull#615]
It doesn't really matter if the integration tests pass or not, because the number of passed vs failed test cases is different with every test run.

Typical example:

```
Testing Time: 1.76s
********************
Failing Tests (6):
    Mull integration tests :: mutations/and_or/01_no_mutations/sample.cpp
    Mull integration tests :: mutations/and_or/03_or_to_and_basic/sample.cpp
    Mull integration tests :: mutations/binary_operators/bitwise/03_right_shift_to_left_shift/sample.cpp
    Mull integration tests :: mutations/binary_operators/relational/02_equal_to_not_equal/sample.cpp
    Mull integration tests :: mutations/negate/03_negate_unary_implicit_cast_pos_to_neg/sample.cpp
    Mull integration tests :: mutations/negate/05_negate_unary_simple_return_neg_to_pos/sample.cpp

  Expected Passes    : 30
  Unexpected Failures: 6
```

```
Failing Tests (7):
    Mull integration tests :: mutations/and_or/01_no_mutations/sample.cpp
    Mull integration tests :: mutations/and_or/03_or_to_and_basic/sample.cpp
    Mull integration tests :: mutations/binary_operators/bitwise/01_no_mutations/sample.cpp
    Mull integration tests :: mutations/binary_operators/bitwise/02_left_shift_to_right_shift/sample.cpp
    Mull integration tests :: mutations/binary_operators/bitwise/03_right_shift_to_left_shift/sample.cpp
    Mull integration tests :: mutations/binary_operators/relational/02_equal_to_not_equal/sample.cpp
    Mull integration tests :: mutations/binary_operators/relational/05_more_than_to_less_than_or_equal/sample.cpp
```

Looking at the Mull's output on the failing test cases reveals that it does not go further than:

```
Stanislaw@home:mull-integration-tests (develop)*$ cat /tmp/wtf.txt
Extracting bitcode from executable --- (threads: 1): 0/1Could not open xar archive
```

This can be directly related to the previous issue #614 and also to the issue reported on the LibEBC here: [ Could not open xar archive #6 ](https://github.com/Guardsquare/LibEBC/issues/6).

**Comment by stanislaw** [comment]
i have tried to run LIT with a number of threads == 1 and I see no issues. Looks like Mull or possible LibEBC use some shared objects, so it is now more clear what to investigate.

**Comment by stanislaw** [comment]
Ok, I was lucky to arrive to a happy end in this debugging session:

The following code was commented our at some point:

```
std::array<std::uint8_t, 16> GenerateUUID() {
  /* uuid_t uuid; */
  /* uuid_generate_random(uuid); */
  std::array<std::uint8_t, 16> data;
  /* std::copy(uuid, uuid + 16, data.begin()); */
  return data;
}
```

According to Alex it was done to provide a quick compatibility with FreeBSD. Also it is a good idea to eliminate any IO from LibEBC long term so let's keep this ticket as a reminder to do that work.

**Comment by stanislaw** [comment]
Has been fixed yesterday by #801.
